docs/reference-guides/rancher-manager-architecture/communicating-with-downstream-user-clusters.md

```diff
@@ -76,6 +76,85 @@ With this endpoint enabled for the downstream cluster, Rancher generates an extr
 
 You will need to use a context defined in this kubeconfig file to access the cluster if Rancher goes down. Therefore, we recommend exporting the kubeconfig file so that if Rancher goes down, you can still use the credentials in the file to access your cluster. For more information, refer to the section on accessing your cluster with [kubectl and the kubeconfig file.](../../how-to-guides/new-user-guides/manage-clusters/access-clusters/use-kubectl-and-kubeconfig.md)
 
+## Impersonation
+
+Users technically exist only on the upstream cluster. When users interact with a downstream cluster through the authentication proxy, there needs to be some entity downstream to serve as the actor for those requests. Rancher creates service accounts to be that entity. Each service account is only granted one permission, which is to **impersonate** the user they belong to. If there was only one service account that could impersonate any user, then it would be possible for a malicious user to corrupt that account and escalate their privileges by impersonating another user. This issue was the basis for a [CVE](https://github.com/rancher/rancher/security/advisories/GHSA-pvxj-25m6-7vqr).
+
+### Impersonation Troubleshooting
+
+On the downstream cluster, five resources handle impersonation:
+
+- namespace: `cattle-impersonation-system`
+- service account: `cattle-impersonation-system/cattle-impersonation-<user ID>`
+- account token secret: `cattle-impersonation-system/cattle-impersonation-<user ID>-token-<hash>`
+- cluster role: `cattle-impersonation-<user ID>`
+- cluster role binding: `cattle-impersonation-<user ID>`
+
+In this example of a typical impersonation cluster role, the system is configured to use `github` as the auth provider:
+
+```
```

Comment:
```suggestion
```yaml
```

I missed that on the first review sorry.  I can see that the highlighting is set to yaml in a lot of other pages so I think it would make sense to have it here as well.